### PR TITLE
Add max mem comparison script and check stderr for message the threashdold was exceeded.

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
@smuzaffar how do I get Jenkins to run the part of the run-pr-comparison script that runs the diff of the maxmem-profile files?

**Comment by gartung** [comment]
Nevermind I found where the files are output
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-04-1100+489fcb/64353/maxmem-comparison/

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by smuzaffar** [comment]
enable gpu

**Comment by smuzaffar** [comment]
please test

File: pr_testing/run-pr-comparisons [hunk 1]
```diff
@@ -392,17 +392,21 @@ jobs
 # maxmem-profile comparison
 # --------------------------------------------------------------------------
 OUTPUT_DIR=$WORKSPACE/results/maxmem-comparison
```
Comment (smuzaffar) [inline pr_testing/run-pr-comparisons:394]: @gartung , please change this to 
```
OUTPUT_DIR=$WORKSPACE/results/maxmem-comparison${TEST_FLAVOR}
```

File: pr_testing/run-pr-comparisons [hunk 2]
```diff
@@ -392,17 +392,21 @@ jobs
...
   WF_FILE=$(basename $WF)
-  #create the output dir
   if [ -f $WORKSPACE/data/$COMPARISON_RELEASE/$WF_DIR/maxmem_profile_${WF_NUMBER}.txt ] ; then
-    diff -u $WORKSPACE/data/PR-${PR_NUM}/$WF_DIR/maxmem_profile_${WF_NUMBER}.txt $WORKSPACE/data/$COMPARISON_RELEASE/$WF_DIR/maxmem_profile_${WF_NUMBER}.txt > $OUTPUT_DIR/${WF_NUMBER}.txt || true
+    $CMS_BOT_DIR/comparisons/compare-maxmem.py $WORKSPACE/data/PR-${PR_NUM}/$WF_DIR/maxmem_profile_${WF_NUMBER}.txt $WORKSPACE/data/$COMPARISON_RELEASE/$WF_DIR/maxmem_profile_${WF_NUMBER}.txt > $OUTPUT_DIR/${WF_NUMBER}.txt 2> $OUTPUT_DIR/${WF_NUMBER}.err || true
   fi
 done
-
+if grep "exceeds threshold" $OUTPUT_DIR/*.err 2>/dev/null; then
+    echo "MAXMEM_COMPARISON${TEST_FLAVOR_STR};OK,max memory used ${UC_TEST_FLAVOR} comparison,See results,/SDT/jenkins-artifacts/$COMP_UPLOAD_DIR/maxmem-comparison" >> ${RESULTS_FILE}
```
Comment (smuzaffar) [inline pr_testing/run-pr-comparisons:406]: also here `/maxmem-comparison` to `/maxmem-comparison${UC_TEST_FLAVOR}`

File: pr_testing/run-pr-comparisons [hunk 3]
```diff
@@ -392,17 +392,21 @@ jobs
...
+else
+    echo "MAXMEM_COMPARISON${TEST_FLAVOR_STR};OK,max memory used ${UC_TEST_FLAVOR} comparison,See results,/SDT/jenkins-artifacts/$COMP_UPLOAD_DIR/maxmem-comparison" >> ${RESULTS_FILE}
```
Comment (smuzaffar) [inline pr_testing/run-pr-comparisons:408]: also here `/maxmem-comparison` to `/maxmem-comparison${UC_TEST_FLAVOR}`

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
An example of a failing threshold check in step 2 of wf 12834.0
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-09-08-0000+489fcb/64446/maxmem-comparison/12834.0.err

**Pull request merged**